onnx_caffe2/backend.py

```diff
@@ -255,6 +258,30 @@ def _create_concat(cls, n):
         op.output.append(dummy_name())
         return op
 
+    @classmethod
+    def _create_optimized_rnn(cls, n):
+        # TODO: we cheat and rely on the fact that ONNX weight layout matches
+        # CuDNN's. Properly we should extract the weight tensor and invoke
+        # RecurrentParamSet exposed by C2
+
+        # TODO: fix Caffe2 to accept initial_h and initial_c as optional inputs
+        assert len(n.inputs) == 4, 'All inputs need to be specified for now'
+        assert len(n.outputs) == 3, 'All outputs need to be specified for now'
+        (w, x, in_h, in_c) = n.inputs
+        (y, out_h, out_c) = n.outputs
+
+        op = core.CreateOperator(
+            'Recurrent',
```

Comment:
Is this operator only available as cudnn op in Caffe2?

Reply:
Yes, currently in C2 it's available on on GPU. We can generate different implementation for CPU (using proper RecurrentNetwork operator with internal cell net), but it won't be a single op. I'll do it separately as a demo how complex subgraphs can be built.